### PR TITLE
Close #1079 add subTitle column and add subTitle in taxon page

### LINKED ISSUE #1079: add sub_title column and add sub_title in taxon page

### PR DESCRIPTION
**Pull request by Sreyleak-Deth**
Screenshot from 2024-02-05 03-06-50

Screenshot from 2024-02-05 03-09-22

### CODE REVIEW

**Review by theachoem — CHANGES_REQUESTED** [review]
Let rename to subtitle instead

**Comment by Sreyleak-Deth** [comment]
> Let rename to subtitle instead

which part b, i can't fund the file that you want me to rename

File: app/overrides/spree/admin/taxons/_form/sub_title.html.erb.deface [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<!--  insert_before "erb[loud]:contains('field_container :permalink')" -->
+
+<div >
+  <%= f.field_container :sub_title do %>
+    <%= f.label :sub_title, raw(Spree.t(:subtitle) + required_span_tag) %>
+    <%= text_field :taxon, :sub_title, class: 'form-control', required: true %>
```
Comment (channainfo) [inline app/overrides/spree/admin/taxons/_form/sub_title.html.erb.deface:6]: Use f.text_field instead. Check the main form to see how to use the text field

**Review by channainfo — CHANGES_REQUESTED** [review]

File: app/overrides/spree/admin/taxons/_form/sub_title.html.erb.deface (hunk 1 above)
Comment (Sreyleak-Deth) [reply app/overrides/spree/admin/taxons/_form/sub_title.html.erb.deface:6]: > Use f.text_field instead. Check the main form to see how to use the text field

Noted bong, I will correct it

Comment (Sreyleak-Deth) [reply app/overrides/spree/admin/taxons/_form/sub_title.html.erb.deface]: > Use f.text_field instead. Check the main form to see how to use the text field

resolved bong

File: app/overrides/spree/admin/taxons/_form/sub_title.html.erb.deface [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<!--  insert_before "erb[loud]:contains('field_container :permalink')" -->
+
+<div>
+  <%= f.field_container :sub_title do %>
```
Comment (theachoem) [inline app/overrides/spree/admin/taxons/_form/sub_title.html.erb.deface:4]: let rename to subtitle

File: app/serializers/spree/v2/storefront/taxon_serializer_decorator.rb [hunk 3]
```diff
@@ -10,7 +10,7 @@ def self.prepended(base)
           base.has_one :web_banner, serializer: ::SpreeCmCommissioner::V2::Storefront::AssetSerializer
           base.has_one :home_banner, serializer: ::SpreeCmCommissioner::V2::Storefront::AssetSerializer
 
-          base.attributes :custom_redirect_url, :kind
+          base.attributes :custom_redirect_url, :kind, :sub_title
```
Comment (theachoem) [inline app/serializers/spree/v2/storefront/taxon_serializer_decorator.rb:13]: rename subtitle as well.

File: config/initializers/spree_permitted_attributes.rb [hunk 4]
```diff
@@ -16,7 +16,9 @@ module PermittedAttributes
       category_icon
       custom_redirect_url
       to_date
-      from_date kind
+      from_date
+      kind
+      sub_title
```
Comment (theachoem) [inline config/initializers/spree_permitted_attributes.rb:21]: rename

File: db/migrate/20240202101246_add_sub_title_to_spree_taxon.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class AddSubTitleToSpreeTaxon < ActiveRecord::Migration[7.0]
+  def change
+    add_column :spree_taxons, :sub_title, :string, if_not_exists: true
```
Comment (theachoem) [inline db/migrate/20240202101246_add_sub_title_to_spree_taxon.rb:3]: rename

File: spec/serializers/spree/v2/storefront/taxon_serializer_spec.rb [hunk 6]
```diff
@@ -41,6 +41,7 @@
         :is_leaf,
         :custom_redirect_url,
         :kind,
+        :sub_title,
```
Comment (theachoem) [inline spec/serializers/spree/v2/storefront/taxon_serializer_spec.rb:44]: rename to subtitle

**Review by theachoem — CHANGES_REQUESTED** [review]

**Pull request merged**